core/main/src/broker/http_broker.rs

```diff
@@ -24,57 +24,80 @@ use ripple_sdk::{
     tokio::{self, sync::mpsc},
     utils::error::RippleError,
 };
+use serde_json::Value;
 
 use super::endpoint_broker::{
     BrokerCallback, BrokerCleaner, BrokerConnectRequest, BrokerOutputForwarder, BrokerRequest,
     BrokerSender, EndpointBroker, EndpointBrokerState, BROKER_CHANNEL_BUFFER_SIZE,
 };
-use crate::state::platform_state::PlatformState;
+use crate::{
+    broker::rules_engine::{jq_compile, RuleTransformType},
+    state::platform_state::PlatformState,
+};
 use tokio_tungstenite::tungstenite::http::uri::InvalidUri;
 
 pub struct HttpBroker {
     sender: BrokerSender,
     cleaner: BrokerCleaner,
 }
-/*
-
-*/
 
 async fn send_http_request(
     client: &Client<HttpConnector>,
-    method: Method,
     uri: &Uri,
-    path: &str,
+    broker_request: BrokerRequest,
 ) -> Result<Response<Body>, RippleError> {
-    /*
-    TODO? we may need to support body for POST request in the future
-    */
-    let http_request = Request::new(Body::empty());
-    let (mut parts, _) = http_request.into_parts();
-    //TODO, need to refactor to support other methods
-    parts.method = method.clone();
-    /*
-    mix endpoint url with method
-    */
-
-    let uri: Uri = format!("{}{}", uri, path)
+    let mut method = Method::GET;
+    let mut body = Body::empty();
+
+    // A rule with a request transform defined indicates that the request is a POST, where
+    // the request transform is the body of the request. Otherwise, it is a GET request.
+
+    if let Some(request_transform) = broker_request
+        .rule
+        .transform
+        .get_transform_data(RuleTransformType::Request)
+    {
+        method = Method::POST;
+
+        let transform_params =
+            match serde_json::from_str::<Vec<Value>>(&broker_request.rpc.params_json) {
+                Ok(mut params) => {
+                    if params.len() > 1 {
```

Comment:
Are the statements on lines 65–68 functionally the same as using

`params.pop().unwrap_or(Value::Null)`

?

Reply:
Yes that's cleaner thanks.

Reply:
Also updated places where I copied that code from.